lib/soap.js

```diff
@@ -120,7 +120,19 @@ SOAP.prototype.parseError = function(body) {
 /** @override **/
 SOAP.prototype.getResponseBody = function(response) {
   var body = SOAP.super_.prototype.getResponseBody.call(this, response);
-  return lookupValue(body, [ /:Envelope$/, /:Body$/, /.+/ ]);
+  if(this._conn.debuggingHeader !== undefined){
+    var Body = lookupValue(body, [ /:Envelope$/, /:Body$/, /.+/ ]); 
+    var Header = lookupValue(body, [ /:Envelope$/, /:Header$/, /.+/ ]); 
+    var newResponse = {
+        body: Body,
+        header: Header
+    };
+    this._conn.debuggingHeader = undefined;
+    return newResponse;
+  }
+  else {
+    return lookupValue(body, [ /:Envelope$/, /:Body$/, /.+/ ]);
+  }
```

Comment:
Switching response type by the connection property is not acceptable because it will cause uncertain effects to the application code. If you want to debug the header information, just add debug log inside it.